metadata-ingestion/src/datahub/ingestion/source/airbyte/rest_api_wrapper/data_resolver.py

```diff
@@ -0,0 +1,440 @@
+            else response[Constant.DESTINATIONNAME],
+        )
+
+    def _get_jobs_endpoint(self, connection_id: str) -> str:
+        jobs_endpoint: str = self.API_ENDPOINTS[Constant.JOBS_LIST]
+        # Replace place holders
+        return jobs_endpoint.format(API_URL=self.api_url, CONNECTION_ID=connection_id)
+
+    def _get_jobs_from_response(self, response: List[Dict]) -> List[Job]:
+        jobs: List[Job] = [
+            Job(
+                job_id=str(job[Constant.JOBID])
+                if Constant.JOBID in job
+                else str(job[Constant.ID]),
+                status=job[Constant.STATUS],
+                job_type=job[Constant.JOBTYPE]
+                if Constant.JOBTYPE in job
+                else job[Constant.CONFIGTYPE],
+                start_time=job[Constant.STARTTIME]
+                if Constant.STARTTIME in job
+                else job[Constant.CREATEDAT],
+                end_time=job[Constant.ENDTIME]
+                if Constant.ENDTIME in job
+                else job[Constant.ENDEDAT],
+                last_updated_at=job[Constant.LASTUPDATEDAT]
+                if Constant.LASTUPDATEDAT in job
+                else job[Constant.UPDATEDAT],
+                bytes_synced=job[Constant.BYTESSYNCED],
+                rows_synced=job[Constant.ROWSSYNCED]
+                if Constant.ROWSSYNCED in job
+                else job[Constant.RECORDSSYNCED],
+            )
+            for job in response
+        ]
+        return jobs
+
+
+class CloudAPIResolver(DataResolverBase):
+    api_url = None
+    # Cloud api endpoints
+    API_ENDPOINTS = {
+        Constant.WORKSPACE_LIST: "{API_URL}/v1/workspaces",
+        Constant.CONNECTION_LIST: "{API_URL}/v1/connections?workspaceIds={WORKSPACE_ID}",
+        Constant.SOURCE_GET: "{API_URL}/v1/sources/{SOURCE_ID}",
+        Constant.DESTINATION_GET: "{API_URL}/v1/destinations/{DESTINATION_ID}",
+        Constant.JOBS_LIST: "{API_URL}/v1/jobs?connectionId={CONNECTION_ID}",
+    }
+
+    def __init__(
+        self,
+        api_url,
+        api_key,
+    ):
+        super().__init__()
+        self.api_url = api_url
+        self.api_key = api_key
+
+    def _get_authorization_header(self) -> Dict:
+        authorization_header = {
+            "User-Agent": "DataHub-RestClient",
+            "accept": "application/json",
+            Constant.AUTHORIZATION: "Bearer {}".format(self.api_key),
+        }
+        return authorization_header
+
+    def test_connection(self):
+        logger.debug(f"Testing connection to api = {self.api_url}")
+        # testing api connection by fetching workspaces
+        response = self._request_session.get(
+            self._get_workspaces_endpoint(),
+            headers=self._get_authorization_header(),
+        )
+        if response.status_code == 401:
+            raise ConfigurationError(
+                "Please check if provided api key is correct or not."
+            )
```

Comment:
@shubhamjagtap639 Please document what minimum permission user or API key needs to fetch metadata

Reply:
There is no any permission thing present at API level